src/main/java/uk/gov/pay/directdebit/mandate/services/OnDemandMandateService.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.pay.directdebit.mandate.services;
+
+import java.time.LocalDate;
+import javax.inject.Inject;
+import javax.ws.rs.core.UriInfo;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GatewayAccount;
+import uk.gov.pay.directdebit.mandate.api.ConfirmMandateRequest;
+import uk.gov.pay.directdebit.mandate.api.CreateMandateRequest;
+import uk.gov.pay.directdebit.mandate.api.CreateMandateResponse;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.mandate.model.MandateType;
+import uk.gov.pay.directdebit.payers.model.AccountNumber;
+import uk.gov.pay.directdebit.payers.model.BankAccountDetails;
+import uk.gov.pay.directdebit.payers.model.SortCode;
+import uk.gov.pay.directdebit.payments.api.CollectPaymentRequest;
+import uk.gov.pay.directdebit.payments.exception.InvalidMandateTypeException;
+import uk.gov.pay.directdebit.payments.model.PaymentProviderFactory;
+import uk.gov.pay.directdebit.payments.model.Transaction;
+import uk.gov.pay.directdebit.payments.services.TransactionService;
+
+import static uk.gov.pay.directdebit.payments.model.DirectDebitEvent.SupportedEvent.DIRECT_DEBIT_DETAILS_CONFIRMED;
+
+public class OnDemandMandateService implements MandateCommandService {
+    private PaymentProviderFactory paymentProviderFactory;
+    private TransactionService transactionService;
+    private MandateStateUpdateService mandateStateUpdateService;
+    private MandateService mandateService;
+
+    @Inject
+    public OnDemandMandateService(
+            PaymentProviderFactory paymentProviderFactory,
+            MandateStateUpdateService mandateStateUpdateService,
+            TransactionService transactionService,
+            MandateService mandateService) {
+        this.paymentProviderFactory = paymentProviderFactory;
+        this.mandateStateUpdateService = mandateStateUpdateService;
+        this.transactionService = transactionService;
+        this.mandateService = mandateService;
+    }
+
+    @Override
+    public void confirm(
```

Comment:
I think it would be nice if we had some uniformity about what service methods returned - i.e. probably the updated object? But no biggie

Reply:
yes, that would've been cool, will se if I can make it work but probably will keep it like this for now :D the PR is quite big